### PR TITLE
[Multi Cluster] Clean up/Streamline + bug fixing

### PR DESCRIPTION
**Pull request by ferantivero**
This PR is just incorporating a few narratives along with some:
- clean up
- streamline 
- bug fixing

solved: #390142; #391002; #368916;#374118;

### CODE REVIEW

File: 04-ca-certificates.md [hunk 1]
```diff
@@ -4,60 +4,54 @@ Now that the [hub-spoke network is provisioned](./03-networking.md), you can fol
...
+   APPGW_FQDN_BU0001A0042_04=$(az deployment group show -g rg-enterprise-networking-spokes -n spoke-BU0001A0042-04 --query properties.outputs.appGwFqdn.value -o tsv)
+   APPGW_SUBDOMAIN_BU0001A0042_03=$(az deployment group show -g rg-enterprise-networking-spokes -n spoke-BU0001A0042-03 --query properties.outputs.subdomainName.value -o tsv)
+   APPGW_SUBDOMAIN_BU0001A0042_04=$(az deployment group show -g rg-enterprise-networking-spokes -n spoke-BU0001A0042-04 --query properties.outputs.subdomainName.value -o tsv)
+   APPGW_IP_RESOURCE_ID_03=$(az deployment group show -g rg-enterprise-networking-spokes -n spoke-BU0001A0042-03 --query properties.outputs.appGatewayPublicIp.value -o tsv)
+   APPGW_IP_RESOURCE_ID_04=$(az deployment group show -g rg-enterprise-networking-spokes -n spoke-BU0001A0042-04 --query properties.outputs.appGatewayPublicIp.value -o tsv)
+
+   # call the Let's Encrypt certificate generation script for both PIP FQDNs
+   sudo chmod +x ./certs/letsencrypt-pip-cert-generation.sh
+   sudo chmod +x ./certs/autogenerator.sh
+   ./certs/letsencrypt-pip-cert-generation.sh $APPGW_SUBDOMAIN_BU0001A0042_03 $APPGW_FQDN_BU0001A0042_03 $APPGW_IP_RESOURCE_ID_03 eastus2
```
Comment (ckittel) [inline 04-ca-certificates.md:32]: Does certbot need to be installed at this point?  (Maybe that was a pre-req that I missed)

File: github-workflow/aks-deploy.yaml [hunk 2]
```diff
@@ -48,6 +48,7 @@ jobs:
           az acr import --source docker.io/library/memcached:1.5.20 -n $ACR_NAME_BU0001A0042 --force
           az acr import --source docker.io/fluxcd/flux:1.19.0 -n $ACR_NAME_BU0001A0042 --force
           az acr import --source docker.io/weaveworks/kured:1.4.0 -n $ACR_NAME_BU0001A0042 --force
+          az acr import --source docker.io/library/traefik:2.2.1 -n $ACR_NAME_BU0001A0042 --force
```
Comment (ckittel) [inline github-workflow/aks-deploy.yaml:51]: Where do you actually use these?  I don't see any `images:` directives in your kustomization files (or `sed` usage) that replace the public container registry usage with these imported images.

**Review by ckittel — CHANGES_REQUESTED** [review]
This is looking really good.  Couple minor suggestions, and a couple questions/observations.  LMK if you want to hop on a call for any of them.

File: 01-prerequisites.md [hunk 3]
```diff
@@ -33,8 +64,8 @@ This is the starting point for the instructions on deploying the [AKS Secure Bas
 1. Fork the repository first, and clone it
 
    ```bash
-   gh repo fork mspnp/aks-secure-baseline --clone=true --remote=false
-   cd aks-secure-baseline
+   gh repo fork mspnp/aks-baseline-multi-region --clone=true --remote=false
```
[the remark this answers is not in the document]
Comment (ferantivero) [reply 01-prerequisites.md:67]: yeah, thanks for checking, I think it is step 6th from pre-requisites section.

[the remark this answers is not in the document]
Comment (ckittel) [reply 01-prerequisites.md:67]: Ah, I see it now, it was in a folded section.  Closing!  Thanks, sorry for the noise.

File: github-workflow/aks-deploy.yaml (hunk 2 above)
Comment (ferantivero) [reply github-workflow/aks-deploy.yaml:51]: I must said this is technical debt at this point. Do you want to work on this? Can we create  AzDO for this instead?

File: 01-prerequisites.md [hunk 4]
```diff
@@ -4,20 +4,51 @@ This is the starting point for the instructions on deploying the [AKS Secure Bas
...
    >
    > - Azure AD [User Administrator](https://docs.microsoft.com/azure/active-directory/users-groups-roles/directory-assign-admin-roles#user-administrator-permissions) is _required_ to create a "break glass" AKS admin Active Directory Security Group and User. Alternatively, you could get your Azure AD admin to create this for you when instructed to do so.
    >   - If you are not part of the User Administrator group in the tenant associated to your Azure subscription, please consider [creating a new tenant](https://docs.microsoft.com/azure/active-directory/fundamentals/active-directory-access-create-new-tenant#create-a-new-tenant-for-your-organization) to use while evaluating this implementation. The Azure AD tenant backing your Cluster's API RBAC does NOT need to be the same tenant associated with your Azure subscription.
 
+   ```bash
+   az login --allow-no-subscriptions -t <Replace-With-ClusterApi-AzureAD-TenantId>
+   ```
+
+   :bulb: The general recommedation here is to use two different tenants for RBAC at the Azure subcription and AKS cluster levels. For more information, please refer to our [AKS Baseline](https://github.com/mspnp/aks-secure-baseline).
```
[the remark this answers is not in the document]
Comment (ferantivero) [reply 01-prerequisites.md]: Yeah, please let me get rid of it. I think it was extracted from regulated as-is it is kind of out of context in here. 

Done | Addressed from [6dca4](https://github.com/mspnp/aks-secure-baseline/pull/184/commits/6dca480b099a71c57feb33357bb021963c72b474)

File: 03-networking.md [hunk 5]
```diff
@@ -19,91 +19,60 @@ The following two resource groups will be created and populated with networking
...
-   ```bash
-   # [This takes less than one minute to run.]
    az group create -n rg-enterprise-networking-spokes -l centralus
    ```
 
-1. Create the two regional network hubs.
+   :bulb: The groups' default location does not matter, as it's not tied to the resource locations. (Please note that typically these resource groups would have already existed.)
 
-   > :book: When the networking team created the regional hubs for eastus2 and westus2, they didn't have any spokes yet defined, yet the networking team always lays out the base hubs following a standard pattern (defined in `hub-default.json`). A hub always contains an Azure Firewall (with some org-wide policies), Azure Bastion, a gateway subnet for VPN connectivity, and Azure Monitor for network observability. They follow Microsoft's recommended sizing for the subnets.
+1. Create two hubs, and two spokes that will be home to the AKS clusters and its adjacent resources and .
```
[the remark this answers is not in the document]
Comment (ferantivero) [reply 03-networking.md]: Good catch, I really wanted to detail our users they also need to enroll the spokes.

Done | Addressed from [2cbe3](https://github.com/mspnp/aks-secure-baseline/pull/184/commits/2cbe3cf7dd3ff3ac8507dac51e4051b88d412da6)

File: 01-prerequisites.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (ckittel) [reply 01-prerequisites.md]: 👍

File: github-workflow/aks-deploy.yaml (hunk 2 above)
Comment (ckittel) [reply github-workflow/aks-deploy.yaml:51]: Go ahead and create a work item for this, and maybe we can get some cycles in the next sprint to address.  Sucks to deploy ACR and never actually use it anywhere or have any instructions on how to do it (other than those buried comments inthe manifest files)

File: 04-ca-certificates.md (hunk 1 above)
Comment (ckittel) [reply 04-ca-certificates.md:33]: Thanks for adding. Resolving.

File: 03-networking.md [hunk 6]
```diff
@@ -19,91 +19,60 @@ The following two resource groups will be created and populated with networking
...
+   az deployment group create -g rg-enterprise-networking-hubs -f networking/hub-region.v1.1.json -n hub-regionB -p nodepoolSubnetResourceIds="['${RESOURCEID_SUBNET_NODEPOOLS_BU0001A0042_04}']" @networking/hub-region.parameters.centralus.json
    ```
 
-   > :book: At this point the networking team has delivered two spokes in which the BU 0001's app team can lay down their AKS clusters. The networking team provides the necessary information to the app teams for them to reference in their Infrastructure-as-Code artifacts.
-   >
-   > Hubs and spokes are controlled by the networking team's GitHub Actions workflows. This automation is not included in this reference implementation as this body of work is focused on the AKS baseline and not the networking team's CI/CD practices.
+   > :bulb: Another interesting use case that this architecture could help with is when AKS introduces _Preview Features_ that could be a breaking in upcoming major releases like happened with `containerd` as the new default runtime. In those situtations, you might want to do some A/B testing without fully disrupting your live and stable AKS cluster.
```
[the remark this answers is not in the document]
Comment (ckittel) [reply 03-networking.md]: Where you moved it is much better than here, IMO.  Thanks!

[the remark this answers is not in the document]
Comment (ferantivero) [reply 03-networking.md]: yeah, good idea. Please let me know.

Moved |  Addressed from [74a1f](https://github.com/mspnp/aks-secure-baseline/pull/184/commits/74a1f43bc76ae3042cf6883b063baa8bab329073)

File: 04-ca-certificates.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ferantivero) [reply 04-ca-certificates.md:33]: done | Addressed from [5866c](https://github.com/mspnp/aks-secure-baseline/pull/184/commits/5866cd67b5dd078d19b23c267c0a72fd70ad71f2)

Comment (ferantivero) [reply 04-ca-certificates.md:33]: added | Addressed from [65a91](https://github.com/mspnp/aks-secure-baseline/pull/184/commits/65a9104e7ef5f6e97caf0774a8cb4424a50ce734)

File: github-workflow/aks-deploy.yaml (hunk 2 above)
Comment (ferantivero) [reply github-workflow/aks-deploy.yaml:51]: done | it is #399511

Comment (ckittel) [reply github-workflow/aks-deploy.yaml:51]: Cool, thanks.

**Review by ckittel — APPROVED** [review]
:shipit: w00t w00t!

**Pull request merged**